AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class EmailValidationService {
+    constructor() {}
+
+    validateEmail(email: string, invalidPattern: string): boolean {
+        /* tslint:disable: max-line-length */
+        const pattern = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.(?:[a-zA-Z0-9](?:\.[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+        const isValidEmail =
+            email &&
+            email.length > 0 &&
+            email.length < 256 &&
+            pattern.test(email.toLowerCase()) &&
+            email.toLowerCase().indexOf(invalidPattern) < 0;
```

Comment:
this.hasCourtroomAccountPattern(email, invalidPattern)

Reply:
This is this.hasCourtroomAccountPattern(email, invalidPattern) used for recognising the judiciary account and courtroom account